applications: serial_lte_modem: Add CONFIG_SLM_CUSTOMIZED flag (#4589)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/slm_description.rst

Check and configure the following configuration options for the sample:

> .. option:: CONFIG_SLM_CUSTOMIZED

Reviewer: ```suggestion
.. option:: CONFIG_SLM_CUSTOMIZED - Flag for customized functionality
```

>     This flag can be used to enable customized functionality

Reviewer: ```suggestion
    This flag can be used to enable customized functionality.
```

> .. option:: CONFIG_SLM_CUSTOMIZED
> 
>    This flag should be enabled when SLM is customized in customer project.
>    Customers should include their own code logic under this flag.
> 
> .. option:: CONFIG_SLM_NATIVE_TLS

Reviewer: ```suggestion
.. option:: CONFIG_SLM_CUSTOMIZED - Flag for customized functionality

   This flag can be used to enable customized functionality.
   To add your own custom logic, add it to the code enclosed by ``#if defined(CONFIG_SLM_CUSTOMIZED)`` and ``#endif /* CONFIG_SLM_CUSTOMIZED */`` and enable this flag.

.. option:: CONFIG_SLM_NATIVE_TLS - Use Zephyr mbedTLS
```

Author: Thanks, the second part changed to below:
  To add your own custom logic, enclose the code by ``#if defined(CONFIG_SLM_CUSTOMIZED)`` and enable this flag.

> .. option:: CONFIG_SLM_NATIVE_TLS

Reviewer: ```suggestion
.. option:: CONFIG_SLM_NATIVE_TLS - Use Zephyr mbedTLS
```
(I know this isn't part of the PR, but it's still wrong in the docs 😉)

Author: Sorry I did not noticed that before, now fixed
